framework/db/ActiveRecord.php

```diff
@@ -199,34 +200,56 @@ protected static function findByCondition($condition)
      * This method will ensure that an array condition only filters on existing table columns.
      *
      * @param array $condition condition to filter.
+     * @param array $aliases
      * @return array filtered condition.
      * @throws InvalidArgumentException in case array contains unsafe values.
+     * @throws InvalidConfigException
      * @since 2.0.15
      * @internal
      */
-    protected static function filterCondition(array $condition)
+    protected static function filterCondition(array $condition, array $aliases)
     {
         $result = [];
         $db = static::getDb();
-        // valid column names are table column names or column names prefixed with table name
+        $columnNames = static::validFilterColumnNames($db, $aliases);
+
+        foreach ($condition as $key => $value) {
+            if (is_string($key) && !in_array($db->quoteSql($key), $columnNames, true)) {
+                throw new InvalidArgumentException('Key "' . $key . '" is not a column name and can not be used as a filter');
+            }
+            $result[$key] = is_array($value) ? array_values($value) : $value;
+        }
+
+        return $result;
+    }
+
+    /**
+     * Valid column names are table column names or column names prefixed with table name or table alias
+     *
+     * @param Connection $db
+     * @param array $aliases
+     * @return array
+     * @throws InvalidConfigException
+     */
+    public static function validFilterColumnNames($db, array $aliases)
```

Comment:
Can we make it protected?

Reply:
The name could be `filterValidColumnNames`.

Reply:
I've  just made the changes. Thanks for the review!